File: 算法篇/YRS1121/最小子数组.py
class Solution:
    def minSubArrayLen(self, s: int, nums: List[int]) -> int:
        # 定义一个无限大的数
        res = float("inf")
        Sum = 0
        index = 0
        for i in range(len(nums)):
            Sum += nums[i]
            while Sum >= s:
                res = min(res, i-index+1)
                Sum -= nums[index]
                index += 1
        if res==float("inf"):
          return 0
        else: 
          return res

File: 算法篇/Wang_Ming/最小子数组.cpp
//算法的时间复杂度O(n)
class Solution {
public:
    int minSubArrayLen(int target, vector<int>& nums) {
        int minLen = INT_MAX;
        int len = 0;
        int sum = 0;
        int slow = 0;
        int fast = 0;
        while (fast < nums.size()){
            if (sum + nums[fast] >= target){
                minLen = min(minLen, len+1);
                
                len--;
                sum -= nums[slow++];
                continue;
            }
            else{
            sum += nums[fast];
            len++;
            fast++;
            }
            
        }
        if (minLen < INT_MAX) return minLen;
        else return 0;
    }
};
